eip-5478.md

```diff
@@ -0,0 +1,49 @@
+---
+eip: 5478
+title: Reducing the gas cost of contract creation with existing code
+author: Qi Zhou (@qizhou)
+discussions-to: https://ethereum-magicians.org/t/eip-5478-reducing-the-gas-cost-of-contract-creation-with-existing-code/10419
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-08-17
+requires: 1014, 2929
+---
+
+## Abstract
+
+Adding a new opcode (CREAET3) that is identical to CREATE2 with potentially much lower gas cost by accepting an additional argument `existing_contract_address` that already stored the code of the new contract.
+
+## Motivation
+
+The EIP aims to reduce the smart contract creation cost of account abstraction (AA) contracts that have identical code.
+
+## Specification
+
+### Parameters
+
+| Constant                  | Value            |
+| ------------------------- | ---------------- |
+| `FORK_BLKNUM`             | TBD              |
+| `CREATE_DATA_GAS_PER_BYTE`          | 200            |
+| `COLD_ACCOUNT_ACCESS_COST` | 2600 |
+| `WARM_ACCOUNT_ACCESS_COST` | 100 |
+
+Adds a new opcode (CREATE3) at 0xf6, which takes 5 stack arguments: `endowment`, `memory_start`, `memory_length`, `salt`, `existing_contract_address`. CREATE3 behaves identically to CREATE2 (0xf5), except that the code hash of the creating contract MUST be the same as that of `existing_contract_address`.
```

Comment:
```suggestion
Adds a new opcode (`CREATE3`) at `0xf6`, which takes 5 stack arguments: `endowment`, `memory_start`, `memory_length`, `salt`, `existing_contract_address`. `CREATE3` behaves identically to `CREATE2` (`0xf5`), except that the code hash of the creating contract MUST be the same as that of `existing_contract_address`.
```

Reply:
Also consider change `0xf6` to an `0x<OPCODE_PH>` placeholder and only specify OPTCODE_PH before a EIP determines the FORK_BLKNUM